### PR TITLE
Support 'more than' and 'less than' statements in verify and try to verify is

### PR DESCRIPTION
**Pull request by peterparser**
This PR adds the support for `more than` and `less than` keywords in `verify` and `try to verify is` statements.

When I was implementing some test cases, I found that this two statements could be useful instead of parsing the data from the kubectl output and do the comparison in a bash fashion.

In case some other changes are needed, please, let me know.

### CODE REVIEW

File: lib/detik.bash [hunk 1]
```diff
@@ -168,7 +168,22 @@ verify() {
...
 		detik_debug ""
 
-		if [[ "$result" == "$card" ]]; then
+		if [[ "$exp" =~ "less" ]]; then
+			if [[ "$result" -lt "$card" ]]; then
+				echo "Found $result $resource named $name (less than $card as expected)."
+			else
+				echo "Found $result $resource named $name (instead of less than $card expected)."
+				return 3
+
```
Comment (vincent-zurczak) [inline lib/detik.bash:177]: Why this empty line here?

File: lib/utils.bash [hunk 2]
```diff
@@ -3,13 +3,17 @@
...
+try_regex_verify_is_more_than="^at +most +([0-9]+) +times +every +([0-9]+)s +to +get +([a-z]+) +named +'([^']+)' +and +verify +that +'([^']+)' +is more than +'([^']+)'$"
+try_regex_verify_is_less_than="^at +most +([0-9]+) +times +every +([0-9]+)s +to +get +([a-z]+) +named +'([^']+)' +and +verify +that +'([^']+)' +is less than +'([^']+)'$"
 try_regex_verify_matches="^at +most +([0-9]+) +times +every +([0-9]+)s +to +get +([a-z]+) +named +'([^']+)' +and +verify +that +'([^']+)' +matches +'([^']+)'$"
 try_regex_find_being="^at +most +([0-9]+) +times +every +([0-9]+)s +to +find +([0-9]+) +([a-z]+) +named +'([^']+)' +with +'([^']+)' +being +'([^']+)'$"
 try_regex_find_matching="^at +most +([0-9]+) +times +every +([0-9]+)s +to +find +([0-9]+) +([a-z]+) +named +'([^']+)' +with +'([^']+)' +matching +'([^']+)'$"
 
 # The regex for the "verify" key word
 verify_regex_count_is="^there +is +(0|1) +([a-z]+) +named +'([^']+)'$"
 verify_regex_count_are="^there +are +([0-9]+) +([a-z]+) +named +'([^']+)'$"
+verify_regex_count_less_than="^there are less than +([0-9]+) +([a-z]+) +named +'([^']+)'$"
```
Comment (vincent-zurczak) [inline lib/utils.bash:15]: Would be better named `verify_regex_count_is_less_than`.

File: lib/utils.bash [hunk 3]
```diff
@@ -3,13 +3,17 @@
...
+verify_regex_count_more_than="^there are more than +([0-9]+) +([a-z]+) +named +'([^']+)'$"
```
Comment (vincent-zurczak) [inline lib/utils.bash:16]: Would be better named `verify_regex_count_is_more_than`.

File: tests/test.detik.try.to.verify.is.bats [hunk 4]
```diff
@@ -224,3 +228,41 @@ mytest_with_spaces() {
...
+}
+
+@test "trying to verify the replicas of a DEPLOYMENT is less than with the lower-case syntax with lower value" {
+	DETIK_CLIENT_NAME=mytest_deployment
+	run try "at most 1 times every 5s to get deployment named 'nginx' and verify that 'status.readyReplicas' is less than '1'"
+	[ "$status" -eq 3 ]
+	[ ${#lines[@]} -eq 2 ]
+	[ "${lines[0]}" = "Valid expression. Verification in progress..." ]
+	[ "${lines[1]}" = "Current value for nginx is not less than 1..." ]
+}
```
Comment (vincent-zurczak) [inline tests/test.detik.try.to.verify.is.bats:268]: Please, add an empty line at the end of the file.

**Review by vincent-zurczak — CHANGES_REQUESTED** [review]
Hi @peterparser!

This is a great contribution.
Thank you!

I added few comments, mostly cosmetic.

**Comment by massimoschembri** [comment]
> Hi @peterparser!
> 
> This is a great contribution. Thank you!
> 
> I added few comments, mostly cosmetic.

Thanks for the review! I've made the fixes and also made the `less/more than` tests more robust.

**Comment by peterparser** [comment]
> Hi @peterparser!
> 
> This is a great contribution. Thank you!
> 
> I added few comments, mostly cosmetic.

Thanks @vincent-zurczak! I've made the changes, also I changed the tests from testing if `more` is present I now check for `more than`, the same happens with `less`. Thanks again

**Comment by brokenpip3** [comment]
hei @peterparser since you are doing this PR can you also add a simple gh action that will run the bats tests? both vs main and during a PR, thanks!

(hoping that is ok for you @vincent-zurczak )

**Review by vincent-zurczak — APPROVED** [review]

**Comment by vincent-zurczak** [comment]
Again, thank you for this contribution.  
I merged it and I was about to create a new tag when I noticed the README documentation had not been updated. Do you think you could do that please?

**Comment by peterparser** [comment]
> @peterparser Again, thank you for this contribution. I merged it and I was about to create a new tag when I noticed the README documentation had not been updated. Do you think you could do that please?

Sure, I will update it and put some examples in the `examples` folder!

**Pull request merged**